FRAuthenticator/FRAuthenticatorTests/UnitTests/RequestInterceptor/RequestInterceptorTests.swift

```diff
@@ -0,0 +1,127 @@
+        // Register RequestInterceptors
+        RequestInterceptorRegistry.shared.registerInterceptors(interceptors: [PushRequestInterceptor()])
+        
+        self.loadMockResponses(["AM_Push_Authentication_Successful"])
+        
+        let qrCode = URL(string: "pushauth://push/forgerock:pushdemouser1?a=aHR0cDovL29wZW5hbS5leGFtcGxlLmNvbTo4MDgxL29wZW5hbS9qc29uL3B1c2gvc25zL21lc3NhZ2U_X2FjdGlvbj1hdXRoZW50aWNhdGU&b=519387&r=aHR0cDovL29wZW5hbS5leGFtcGxlLmNvbTo4MDgxL29wZW5hbS9qc29uL3B1c2gvc25zL21lc3NhZ2U_X2FjdGlvbj1yZWdpc3Rlcg&s=O9JHEGfOsaZqc5JT0DHM5hYFA8jofohw5vAP0EpG4JU&c=75OQ3FXmzV99TPf0ihevFfB0s43XsxQ747sY6BopgME&l=YW1sYmNvb2tpZT0wMQ&m=REGISTER:fe6311ab-013e-4599-9c0e-4c4e2525199b1588721418483&issuer=Rm9yZ2VSb2NrU2FuZGJveA")!
+        
+        do {
+            let parser = try PushQRCodeParser(url: qrCode)
+            let mechanism = PushMechanism(issuer: parser.issuer, accountName: parser.label, secret: parser.secret, authEndpoint: parser.authenticationEndpoint, regEndpoint: parser.registrationEndpoint, messageId: parser.messageId, challenge: parser.challenge, loadBalancer: parser.loadBalancer)
+            mechanism.mechanismUUID = "32E28B44-153C-4BDE-9FDB-38069BC23D9C"
+            FRAClient.storage.setMechanism(mechanism: mechanism)
+            
+            let messageId = "AUTHENTICATE:8af40ee6-8fa0-4bdd-949c-1dd29d5e55931588721432364"
+            var notificationPayload: [String: String] = [:]
+            notificationPayload["c"] = "6ggPLysKJ6wSwBsQFtPclHQKebpOTMNwHP53kZxIGE4="
+            notificationPayload["t"] = "120"
+            notificationPayload["u"] = "32E28B44-153C-4BDE-9FDB-38069BC23D9C"
+            notificationPayload["l"] = "YW1sYmNvb2tpZT0wMQ=="
+            
+            let ex = self.expectation(description: "PushNotification Authentication")
+            let notification = try PushNotification(messageId: messageId, payload: notificationPayload)
+            notification.accept(onSuccess: {
+                // Expected to be successful
+                ex.fulfill()
+            }) { (error) in
+                XCTFail("Push authentication failed while expecting to be successful")
+                ex.fulfill()
+            }
+            waitForExpectations(timeout: 60, handler: nil)
+            
+            XCTAssertEqual(RequestInterceptorTests.intercepted.count, 1)
+            let interceptorsInOrder: [String] = ["PUSH_AUTHENTICATE"]
+            for (index, intercepted) in RequestInterceptorTests.intercepted.enumerated() {
+                XCTAssertEqual(interceptorsInOrder[index], intercepted)
+            }
+        }
+        catch {
+            XCTFail("Push authentication failed to prepare auth request")
+        }
+    }
+}
+
+class PushRequestInterceptor: RequestInterceptor {
+    func intercept(request: Request, action: Action) -> Request {
+        if action.type == "PUSH_REGISTER" {
+            RequestInterceptorTests.intercepted.append("PUSH_REGISTER")
```

Comment:
as an extra step , i think we can add some custom header add those custom header part of AM_Push_Registration_Successful mocks to make sure the request has custom header .

Reply:
Updated the tests. added custom header to the request.